src/services/LeafValueCoder.ts

```diff
@@ -22,6 +33,20 @@ export class LeafValueCoder {
   /**
    *
    * @param leafAsHex {string} data in hex format
+   * @param label {string}
    */
-  static decode = (leafAsHex: string): number => new BigNumber(leafAsHex, 16).div(NUMERIC_MULTIPLIER).toNumber();
+  static decode = (leafAsHex: string, label: string): number | string => {
+    const bn = new BigNumber(leafAsHex, 16);
+    return LeafValueCoder.isFixedValue(label) ? bn.toFixed() : bn.div(NUMERIC_MULTIPLIER).toNumber();
+  };
+
+  static isFixedValue = (label: string): boolean => label.startsWith(FIXED_NUMBER_PREFIX);
```

Comment:
how does this work? we're using keys to detect data type now?

Reply:
yes